doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,299 @@
+operators with :meth:`.fit`, :meth:`.transform` (and :meth:`.predict`)
+capability support pandas in pandas out. However, there are ways around this
+such as an example given
+`here <https://github.com/scikit-learn/scikit-learn/issues/5523#issuecomment-171674105>`__
+show, where adding additional functionality to the StandardScaler class adds
+the pandas in pandas out capability. Care should be taken as this does not
+take care of the column ordering problem that is discussed in the next section.
+
+The column ordering problem
+===========================
+
+Because Scikit-learn transforms DataFrames to NumPy arrays, it should be
+assumed, that all information and benefits of column names is lost and that
+from that point forward, only column order and not column labels stay relevant.
+This can cause problems when e.g. pickling a trained estimator and later
+applying it to a new DataFrame that, while having the same data columns and
+labels, has those in a different order compared to the original DataFrame.
+Intuitively it might be assumed that because Scikit-learn handles the use of
+DataFrames so smoothly in most cases, the same goes for re-ordering labeled
+DataFrames but this is **not** the case.
+
+An example of how this might impact your future prediction can be seen in the
+example given below (original with slight modifications adjusting for current
+API, thanks to `SauceCat <https://github.com/scikit-learn/scikit-learn/issues/7242#issue-173131995>`__).
+
+>>> # for simplification, consider a very simple case
+>>> from sklearn.datasets import load_iris
+>>> import pandas as pd
+>>>
+>>> # make a dataframe
+>>> iris = load_iris()
+>>> X, y = iris.data[:-1,:], iris.target[:-1]
+>>> iris_pd = pd.DataFrame(X)
+>>> iris_pd.columns = iris.feature_names
+>>> iris_pd['target'] = y
+>>>
+>>> from sklearn.model_selection import train_test_split
+>>> train, test = train_test_split(iris_pd, test_size= 0.3, random_state=42)
+>>>
+>>> feature_columns_train = ['sepal length (cm)','sepal width (cm)',
+...                          'petal length (cm)','petal width (cm)']
+>>> # last two correct order
+>>> feature_columns_test = ['sepal length (cm)','sepal width (cm)',
+...                         'petal width (cm)','petal length (cm)']
+>>> # last two switched order
+>>>
+>>> from sklearn.linear_model import LogisticRegression
+>>> lg = LogisticRegression(n_jobs=4, random_state=123, verbose=0,
+...                         penalty='l2', C=1.0,
+...                         solver='lbfgs', multi_class='auto')
+>>> lg.fit(train[feature_columns_train], train['target'])
+LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
+                   intercept_scaling=1, l1_ratio=None, max_iter=100,
+                   multi_class='auto', n_jobs=4, penalty='l2', random_state=123,
+                   solver='lbfgs', tol=0.0001, verbose=0, warm_start=False)
+>>>
+>>> prob1 = lg.predict_proba(test[feature_columns_train])
```

Comment:
can you show the output of `predict` instead of `predict_proba` here? (I assume that will be easier to interpret)